panther_manager/include/panther_manager/plugins/shutdown_host.hpp

```diff
@@ -0,0 +1,251 @@
+#ifndef PANTHER_MANAGER_SHUTDOWN_HOST_HPP_
+#define PANTHER_MANAGER_SHUTDOWN_HOST_HPP_
+
+#include <cstdlib>
+#include <functional>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <libssh/libssh.h>
+
+#include <ros/time.h>
+
+namespace panther_manager
+{
+
+enum class ShutdownHostState {
+  IDLE,
+  COMMAND_EXECUTED,
+  RESPONSE_RECEIVED,
+  PINGING,
+  SKIPPED,
+  SUCCESS,
+  FAILURE,
+};
+
+class Host
+{
+public:
+  Host(const std::string ip, const std::string user, const int port = 22)
+  : ip_(ip),
+    user_(user),
+    port_(port),
+    hash_(std::hash<std::string>{}(ip + user + std::to_string(port)))
+  {
+  }
+
+  std::string get_ip() const { return ip_; }
+  std::string get_user() const { return user_; }
+  int get_port() const { return port_; }
+
+  bool is_available() const
+  {
+    return system(("ping -c 1 -w 1 " + ip_ + " > /dev/null").c_str()) == 0;
+  }
+
+  bool operator==(const Host & other) const { return hash_ == other.hash_; }
+
+  bool operator!=(const Host & other) const { return hash_ != other.hash_; }
+
+  bool operator<(const Host & other) const { return hash_ < other.hash_; }
+
+private:
+  const std::string ip_;
+  const std::string user_;
+  const std::size_t hash_;
+  const int port_;
+};
+
+class ShutdownHost : public Host
+{
+public:
+  // default constructor
+  ShutdownHost() : Host("", ""), command_(""), timeout_(5.0), ping_for_success_(true) {}
+  ShutdownHost(
+    const std::string ip, const std::string user, const int port = 22,
+    const std::string command = "sudo shutdown now", const float timeout = 5.0,
+    const bool ping_for_success = true)
+  : Host(ip, user, port),
+    command_(command),
+    timeout_(timeout),
+    ping_for_success_(ping_for_success),
+    state_(ShutdownHostState::IDLE)
+  {
+  }
+
+  ~ShutdownHost() {}
+
+  std::string get_error() const { return failure_reason_; }
+  std::string get_response() const { return output_; }
+  ShutdownHostState get_state() const { return state_; }
+
+  void call()
+  {
+    switch (state_) {
+      case ShutdownHostState::IDLE:
+        if (!is_available()) {
+          state_ = ShutdownHostState::SKIPPED;
+          break;
+        }
+
+        try {
+          request_shutdown();
+        } catch (std::runtime_error err) {
+          state_ = ShutdownHostState::FAILURE;
+          failure_reason_ = err.what();
+          break;
+        }
+        state_ = ShutdownHostState::COMMAND_EXECUTED;
+        break;
+
+      case ShutdownHostState::COMMAND_EXECUTED:
+        try {
+          if (update_response()) break;
```

Comment:
```suggestion
          if (update_response()) {
              break;
          }
```
Just for the sake of standardization